python/lsst/pipe/tasks/measurementDriver.py

```diff
@@ -0,0 +1,379 @@
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+__all__ = ["MeasurementDriverConfig", "MeasurementDriverTask"]
+
+import logging
+
+import lsst.afw.image as afwImage
+import lsst.afw.table as afwTable
+import lsst.meas.algorithms as measAlgorithms
+import lsst.meas.base as measBase
+import lsst.meas.deblender as measDeblender
+import lsst.meas.extensions.scarlet as scarlet
+import lsst.pex.config as pexConfig
+import lsst.pipe.base as pipeBase
+import numpy as np
+
+logging.basicConfig(level=logging.INFO)
+
+
+class MeasurementDriverConfig(pexConfig.Config):
+    """Configuration parameters for `MeasurementDriverTask`."""
+
+    # To generate catalog ids consistently across subtasks.
+    id_generator = measBase.DetectorVisitIdGeneratorConfig.make_field()
+
+    detection = pexConfig.ConfigurableField(
+        target=measAlgorithms.SourceDetectionTask,
+        doc="Task to detect sources to return in the output catalog.",
+    )
+
+    deblender = pexConfig.ChoiceField[str](
+        doc="The deblender to use.",
+        default="meas_deblender",
+        allowed={"meas_deblender": "Deblend using meas_deblender", "scarlet": "Deblend using scarlet"},
+    )
+
+    deblend = pexConfig.ConfigurableField(
+        target=measDeblender.SourceDeblendTask, doc="Split blended sources into their components."
+    )
+
+    measurement = pexConfig.ConfigurableField(
+        target=measBase.SingleFrameMeasurementTask,
+        doc="Task to measure sources to return in the output catalog.",
+    )
+
+    def __setattr__(self, key, value):
+        """Intercept attribute setting to trigger setDefaults when relevant
+        fields change.
+        """
+        super().__setattr__(key, value)
+
+        # This is to ensure the deblend target is set correctly whenever the
+        # deblender is changed. This is required because `setDefaults` is not
+        # automatically invoked during reconfiguration.
+        if key == "deblender":
+            self.setDefaults()
+
+    def validate(self):
+        super().validate()
+
+        # Ensure the deblend target aligns with the selected deblender.
+        if self.deblender == "scarlet":
+            assert self.deblend.target == scarlet.ScarletDeblendTask
+        elif self.deblender == "meas_deblender":
+            assert self.deblend.target == measDeblender.SourceDeblendTask
+        elif self.deblender is not None:
+            raise ValueError(f"Invalid deblender value: {self.deblender}")
+
+    def setDefaults(self):
+        super().setDefaults()
+        if self.deblender == "scarlet":
+            self.deblend.retarget(scarlet.ScarletDeblendTask)
+        elif self.deblender == "meas_deblender":
+            self.deblend.retarget(measDeblender.SourceDeblendTask)
```

Comment:
Would it be better to rename this method to `setDeblender()` and just call it in the `setDefaults` method? This way my`__setattr__` method becomes:
```python
def __setattr__(self, key, value):
    super().__setattr__(key, value)
    if key == "deblender":
        self.setDeblender()
```
It just makes more sense to me this way.

Reply:
See my comment about why I used both `deblend` and `deblender` in the config.